pkg/server/server.go

```diff
@@ -185,7 +185,12 @@ func NewProxyServer(serverID string, serverCount int, agentAuthenticationOptions
 
 // Proxy handles incoming streams from gRPC frontend.
 func (s *ProxyServer) Proxy(stream client.ProxyService_ProxyServer) error {
-	klog.Info("proxy request from client")
+	md, ok := metadata.FromIncomingContext(stream.Context())
+	if !ok {
+		return fmt.Errorf("failed to get context")
+	}
+	userAgent := md.Get("user-agent")
```

Comment:
As mentioned by Adel, this should be "User-Agent". RFC 7231 specifies it this was and as a parameter name it would be case sensitive.

Reply:
GRPC metadata pairs are normalized as lowercase strings. (https://github.com/grpc/grpc-go/blob/master/metadata/metadata.go#L110). The examples I looked at used lowercase representations for "user-agent" and I tried to follow that.

Reply:
Goo point. As long as we're not setting it in the HTTP headers, then this should be correct.

Reply:
Also can you define it in the headers package?